### PR TITLE
[denonmarantz] Fix NPE on connection problem

### LINKED ISSUE #15444: [denonmarantz] Cannot invoke BufferReader syslog
Raspberry Pi 3 B
Openhab 4.0.2-1 -> also with 4.0.1

I investigate an unstable issue with openhab and i investigate in all directions. Somebody say that following errors in syslog aren't good that i have to switch off the binding. But i need this binding. 

If the receiver isn't available via network i get this issue in syslog:

`Aug 17 15:58:02 openhabian karaf[24451]: #011at org.openhab.binding.denonmarantz.internal.connector.telnet.DenonMarantzTelnetClientThread.run(DenonMarantzTelnetClientThread.java:68)
Aug 17 15:58:36 openhabian karaf[24451]: Exception in thread "OH-binding-denonmarantz:avr:2b74ec1a" java.lang.NullPointerException: Cannot invoke "java.io.BufferedReader.readLine()" because "this.in" is null`

Thanks.

### PR DESCRIPTION
**Pull request by lsiepel**
- Fixes #15444 
- Checkstyle cleanup

Test jar: https://1drv.ms/u/s!AnMcxmvEeupwjqVmDjS6MTGKtfRFQQ?e=orDdgn

### CODE REVIEW

**Review by holgerfriedrich — APPROVED** [review]
Lgtm

File: bundles/org.openhab.binding.denonmarantz/src/main/java/org/openhab/binding/denonmarantz/internal/connector/telnet/DenonMarantzTelnetClientThread.java [hunk 1]
```diff
@@ -65,7 +66,10 @@ public void run() {
 
             do {
                 try {
-                    String line = in.readLine();
+                    String line = null;
+                    if (in != null) {
+                        line = in.readLine();
```
Comment (holgerfriedrich) [inline bundles/org.openhab.binding.denonmarantz/src/main/java/org/openhab/binding/denonmarantz/internal/connector/telnet/DenonMarantzTelnetClientThread.java:71]: This seems to cause a new compiler warning, '''DenonMarantzTelnetClientThread.java:[71,32] Potential null pointer access: this expression has a '@Nullable' type'''.

Although there should be only one thread, you will probably need a local copy of '''in''' to get rid of this message.

**Review by holgerfriedrich — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.denonmarantz/src/main/java/org/openhab/binding/denonmarantz/internal/connector/telnet/DenonMarantzTelnetClientThread.java (hunk 1 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.denonmarantz/src/main/java/org/openhab/binding/denonmarantz/internal/connector/telnet/DenonMarantzTelnetClientThread.java:72]: Fixed.

**Review by holgerfriedrich — APPROVED** [review]
Lgtm

**Review by jlaur — APPROVED** [review]
Thanks for the fix - LGTM!

I can also confirm that it works. I was able to reproduce the NPE from the 4.0.2 version, and with the fix the Thing will correctly transition between ONLINE and OFFLINE when turning the receiver on/off.

**Pull request merged**